### PR TITLE
CI: inline the move-caches script into the action

### PR DESCRIPTION
**Pull request by erik-krogh**
That way the action also works when used by another repository that includes `github/codeql` as a submodule.  

And a drive-by change that enforces that we always run QL-for-QL.

### CODE REVIEW

**Comment by kaeluka** [comment]
> That way the action also works when used by another repository that includes github/codeql as a submodule.

How is that currently broken?

File: .github/actions/cache-query-compilation/action.yml [hunk 1]
```diff
@@ -43,12 +43,83 @@ runs:
...
         COMBINED_CACHE_DIR: ${{ runner.temp }}/compilation-dir
+      with:
+        script: |
+          // # Move all the existing cache into another folder, so we only preserve the cache for the current queries.
+          // mkdir -p ${COMBINED_CACHE_DIR}
+          // rm -f **/.cache/{lock,size} # -f to avoid errors if the cache is empty.
+          // # copy the contents of the .cache folders into the combined cache folder.
+          // cp -r **/.cache/* ${COMBINED_CACHE_DIR}/ || : # ignore missing files
+          // # clean up the .cache folders
+          // rm -rf **/.cache/*
```
Comment (kaeluka) [inline .github/actions/cache-query-compilation/action.yml:57]: delete?

```suggestion
```

**Comment by erik-krogh** [comment]
> How is that currently broken?

The `$GITHUB_WORKSPACE` path references the "root" repository.  
So the path to `move-caches.js` didn't work when `github/codeql` is within a submodule.  
I didn't find a good way to get the path to the current workflow file (well, I didn't find anything that worked for composite workflows).
So inlining the script seemed like a good solution.

File: .github/workflows/ql-for-ql-build.yml [hunk 2]
```diff
@@ -5,13 +5,6 @@ on:
     branches: [main]
   pull_request:
     branches: [main]
-    paths:
```
Comment (kaeluka) [inline .github/workflows/ql-for-ql-build.yml:8]: why remove these paths?

**Review by kaeluka — APPROVED** [review]
left two minor comments

File: .github/workflows/ql-for-ql-build.yml (hunk 2 above)
Comment (erik-krogh) [reply .github/workflows/ql-for-ql-build.yml:8]: I wanted to trigger QL-for-QL on this PR. 
The simplest approach was just to always run QL-for-QL.

**Pull request merged**